monai/apps/auto3dseg/ensemble_builder.py

```diff
@@ -327,3 +335,225 @@ def get_ensemble(self):
         """Get the ensemble"""
 
         return self.ensemble
+
+class EnsembleRunner:
+    """
+    The Runner for ensembler
+
+    Args:
+        work_dir: working directory to save the intermediate and final results.
+        data_src_cfg_name: filename of the data source.
+        num_fold: number of fold.
+        ensemble_method_name: method to ensemble predictions from different model.
+                              Suported methods: ["AlgoEnsembleBestN", "AlgoEnsembleBestByFold"].
+        mgpu: if using multi-gpu.
+        kwargs: additional image writing, ensembling parameters and prediction parameters for the ensemble inference.
+    Examples:
+
+        .. code-block:: python
+
+            ensemble_runner = EnsembleRunner(data_src_cfg_name,
+                                             work_dir,
+                                             ensemble_method_name,
+                                             mgpu=device_setting['n_devices']>1,
+                                             **kwargs,
+                                             **pred_params)
+            ensemble_runner.run(device_setting)
+
+    """
+    def __init__(self, data_src_cfg_name: str='./work_dir/input.yaml',
+                 work_dir: str='./work_dir',
+                 num_fold: int=5,
+                 ensemble_method_name: str='AlgoEnsembleBestByFold',
+                 mgpu: bool=True,
+                 **kwargs):
```

Comment:
We can add an optional argument `transform=None` here, and accept the save transform as one of the post-processing step.

Reply:
Hi @heyufan1995 ! Although you mentioned that we can only pass strings, instead of an instance such as SaveImage Transforms, I have a new idea and it is open to discussion.

- Python fire API supports dictionary, e.g. `python example.py {"name":'"David Bieber"'}`
- MONAI ConfigParser can resolve a dictionary and instantiate a class

So it may be possible combine both, to construct a dictionary in `set_image_save_transform`, and pass that dictionary to EnsembleRunner. The instantiation of SaveImage Transform can be inside the EnsembleRunner as you want, and we don't need to discontinue the API of `set_image_save_transform`.

Reply:
Feel free to close this one as we addressed the API breaking change.